src/osp/core/array_view.h

```diff
@@ -28,10 +28,56 @@
 #include <Corrade/Containers/ArrayView.h>
 // IWYU pragma: end_exports
 
+#include <iterator>
+
 namespace osp
 {
 
 using Corrade::Containers::ArrayView;
 using Corrade::Containers::arrayView;
 
+template<typename T>
+struct ArrayView2DWrapper
+{
+    constexpr ArrayView<T> row(std::size_t const index) const noexcept
+    {
+        return view.sliceSize(index * columns, columns);
+    }
+
+    ArrayView<T> view;
+    std::size_t columns;
+};
+
+template<typename T>
+constexpr ArrayView2DWrapper<T> as_2d(ArrayView<T> view, std::size_t columns) noexcept
+{
+    return { .view = view, .columns = columns };
+}
+
+/**
+ * @brief slice_2d_row
+ * @param view
+ * @param index
+ * @param size
+ */
+template<typename T>
+ArrayView<T> slice_2d_row(ArrayView<T> const& view, std::size_t index, std::size_t size) noexcept
+{
+    return view.sliceSize(index, size);
+}
+
+/**
+ * @brief Anger the address sanitizer for invalid array views / spans / similar containers
+ */
+template<typename T>
+void debug_touch_container(T const& container)
+{
+    // This feels kinda stupid, but if you're reading this and have a better idea then go ahead
```

Comment:
ydu're -> you're

Reply:
I don't see any mentions of "ydu're"? is there something changing text on your end?

Reply:
hrmm, my computer was running updates, and i guess replaced a DLL firefox wanted, and it spazzed out.

Ignore, problem on my end.